CMakeLists.txt

```diff
@@ -7,19 +7,86 @@ if (NOT SKBUILD)
 endif()
 
-# {{{ Constrain FindPython to find the Python version used by scikit-build
+# {{{ Options
+
+option(PYOPENCL_TRACE "Enable OpenCL tracing" OFF)
+option(PYOPENCL_ENABLE_GL "Enable OpenGL interoperability" OFF)
+option(PYOPENCL_USE_SHIPPED_EXT "Use shipped CL extension header" OFF)
+set(CL_INC_DIR CACHE STRING "OpenCL include directory")
+set(CL_LIB_DIR CACHE STRING "OpenCL library directory")
+set(CL_LIBNAME CACHE STRING "OpenCL library name")
+
+if(NOT CL_INC_DIR)
+  message(STATUS "CL_INC_DIR not set, trying to guess it from conda environment and environment variables")
+  if(DEFINED ENV{CONDA_PREFIX})
+    if(EXISTS $ENV{CONDA_PREFIX}/lib/libOpenCL${CMAKE_SHARED_LIBRARY_SUFFIX})
+      message(STATUS "Found OpenCL in conda environment '$ENV{CONDA_PREFIX}'")
+      set(CL_INC_DIR $ENV{CONDA_PREFIX}/include)
+      set(CL_LIB_DIR $ENV{CONDA_PREFIX}/lib)
+      set(CL_LIBNAME OpenCL)
+    endif()
+  endif()
+
+  if(DEFINED ENV{CL_INC_DIR})
+    message(STATUS "Using OpenCL include directory from environment '$ENV{CL_INC_DIR}'")
+    set(CL_INC_DIR $ENV{CL_INC_DIR})
+  endif()
+
+  if(DEFINED ENV{CL_LIB_DIR})
+    message(STATUS "Using OpenCL library directory from environment '$ENV{CL_INC_DIR}'")
+    set(CL_LIB_DIR $ENV{CL_LIB_DIR})
+  endif()
+
+  if(DEFINED ENV{CL_LIBNAME})
+    message(STATUS "Using OpenCL library name from environment '$ENV{CL_LIBNAME}'")
+    set(CL_LIBNAME $ENV{CL_LIBNAME})
+  endif()
+endif(NOT CL_INC_DIR)
+
+message(STATUS "CL_INC_DIR ${CL_INC_DIR}")
+message(STATUS "CL_LIB_DIR ${CL_LIB_DIR}")
+message(STATUS "CL_LIBNAME ${CL_LIBNAME}")
+
+# }}}
 
-if (SKBUILD)
-  message(STATUS "Python_VERSION ${PYTHON_VERSION_STRING}")
-  message(STATUS "Python_EXECUTABLE ${PYTHON_EXECUTABLE}")
-  message(STATUS "Python_INCLUDE_DIR ${PYTHON_INCLUDE_DIR}")
-  message(STATUS "Python_LIBRARIES ${PYTHON_LIBRARY}")
-  set(Python_VERSION "${PYTHON_VERSION_STRING}")
-  set(Python_EXECUTABLE "${PYTHON_EXECUTABLE}")
-  set(Python_INCLUDE_DIR "${PYTHON_INCLUDE_DIR}")
-  set(Python_LIBRARIES "${PYTHON_LIBRARY}")
+# {{{ Get version information
+
+find_program(GIT git)
+
+if(GIT AND EXISTS ${CMAKE_SOURCE_DIR}/.git)
+  execute_process(COMMAND git describe --exact-match --dirty=*
+             OUTPUT_VARIABLE PYOPENCL_VERSION_GIT
+             RESULT_VARIABLE git_result
+             OUTPUT_STRIP_TRAILING_WHITESPACE
+             ERROR_QUIET
+             )
+  if(NOT ${git_result} EQUAL 0)
+    execute_process(COMMAND git describe --long --always --dirty=*
+             OUTPUT_VARIABLE PYOPENCL_VERSION_GIT
+             OUTPUT_STRIP_TRAILING_WHITESPACE
+             )
+    set(PYOPENCL_REL "(dev)")
+  else()
+    set(PYOPENCL_REL "(release)")
+  endif()
+else()
+  set(PYOPENCL_VERSION_GIT "v${PROJECT_VERSION}")
 endif()
-find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)
+
+# }}}
+
+# {{{ Constrain FindPython to find the Python version used by scikit-build
+
+message(STATUS "Python_VERSION ${PYTHON_VERSION_STRING}")
```

Comment:
Who sets `PYTHON_VERSION_STRING` and these over variables? `scikit-build-core`?

Reply:
Good question, as far as I can tell, this part of the code isn't functional (e.g., `PYTHON_VERSION_STRING` isn't set at all). Removed in 08add61.